src/threading.c

```diff
@@ -184,9 +186,14 @@ void jl_init_threading(void)
         jl_n_threads = jl_max_threads;
 
     // set up space for per-thread heaps
-    jl_all_heaps = malloc(jl_n_threads * sizeof(void*));
-    jl_all_pgcstacks = malloc(jl_n_threads * sizeof(void*));
-    jl_all_task_states = malloc(jl_n_threads * sizeof(jl_thread_task_state_t));
+    jl_all_heaps =
+#ifdef __cplusplus
+        // undeclared?
```

Comment:
you need to either go with `struct _jl_thread_heap_t` or `jl_thread_heap_t`, not half way in-between.